### ISSUE TITLE
Evaluate switching globaleaks repository to packagecloud

### ISSUE THREAD

**Issue by fpietrosanti** [globaleaks/GlobaLeaks#2603]
**Current behavior**
Package cloud provide very valuable facilities to maintain a repository https://packagecloud.io/l/apt-repository including automatic pushing of debian packages from CI-systems, TLS, GPG, IPv6 and based on their approval can provide also a free plan for opensource projects.

In order to reduce the system maintenance we could consider switching globaleaks repository to such a hosted cloud system.

**Comment by evilaliv3** [comment]
Actually the mantainance of our private repository is not that demanding, it just require periodic update of the serve like for a normal system.

If we want anyway to consider an alternative maybe we could just consider to create an Ubuntu PPA?

P.s.: currently we are not hosting only Globaleaks but as well many dependencies related to up-to-date versions of libsodium, tor, acme, etc.

**Comment by fpietrosanti** [comment]
@evilaliv3 the value imho is not the cloud repository itself, offering major resiliency over self-managed system, but is the CI and CD cloud pipeline that could be in a cloudified-way setup that's interesting, see https://packagecloud.io/docs#ci where from travis-ci could already create per-commit package and then push it for continuous deployment testing to be deployed trough https://buildkite.com/features

**Comment by evilaliv3** [comment]
I agree, this all makes sense. Let's evaluate well all the alternatives with long term vision.

Have you already evaluated if packagecloud does allow to load other dependencies as well (implementing the backport of packages that we are doing for example for Tor/Libsodium/Acme?)

**Comment by fpietrosanti** [comment]
Ref #2605

**Comment by lskillen** [comment]
Hey @evilaliv3 @fpietrosanti Lee from @cloudsmith-io here. If you're looking for a top-tier package management service that also offers the ability to have custom distribution domains (e.g. `dl.globaleaks.org` or `pkg.globaleaks.org`, etc.), please check out [Cloudsmith](https://cloudsmith.io).

Disclosure: I work there, but help companies and individuals to do this kind-of thing each day. :-)
